### PR TITLE
Version: 3.36.1

### PR DESCRIPTION
**Pull request by kira-syslogng**
TODO:

- [x] News entry from #3929
- [x] News entry from #3911
- [x] Double check `versioning.h` bump (pre4.0 additions)
- [x] #3903
- [x] #3927 
- [x] #3885

Tarball is available at: https://github.com/syslog-ng/syslog-ng/actions/runs/1910720449

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: NEWS.md [hunk 1]
```diff
@@ -1,133 +1,203 @@
...
+   sends a correct command, but anyone with access to the UNIX domain socket
+   `syslog-ng.ctl` (root only by default) can trigger a crash.
+   ([#3900](https://github.com/syslog-ng/syslog-ng/pull/3900))
+ * `credit-card-mask()`: fixed visa, mastercard and jcb card regex pattern
+   ([#3853](https://github.com/syslog-ng/syslog-ng/pull/3853))
+ * `cisco-parser()`: allow a leading dot in the timestamp (not synced clocks)
+   ([#3843](https://github.com/syslog-ng/syslog-ng/pull/3843))
+
+   A completely filled and then emptied disk-buffer may have been recognised as corrupt.
+   ([#3874](https://github.com/syslog-ng/syslog-ng/pull/3874))
```
Comment (MrAnno) [inline NEWS.md:181]: This belongs to "`disk-buffer()`: fixed a disk-buffer corruption issue".
Is this a bug in the news generator script?

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: NEWS.md (hunk 1 above)
Comment (alltilla) [reply NEWS.md]: No, I messed it up by hand, sorry. Fixed.

Comment (alltilla) [reply NEWS.md]: Nice catch btw!

File: NEWS.md [hunk 2]
```diff
@@ -1,133 +1,203 @@
...
+   RFC5424 (e.g.  `syslog()` source).
+   ([#3893](https://github.com/syslog-ng/syslog-ng/pull/3893))
+ * `transport(text-with-nuls)`: added a new transport mechanism for
+   the `network()` driver that allows `NUL` characters within the message.
+
+   Note: syslog-ng does not support embedded `NUL` characters everywhere, so it is
+   recommended that you also use `flags(no-multi-line)` that causes `NUL`
+   characters to be replaced by space.
+   ([#3913](https://github.com/syslog-ng/syslog-ng/pull/3913))
+ * `java`: upgraded from old log4j v1.x line to log4j v2.17.2
```
Comment (MrAnno) [inline NEWS.md:133]: I'd list this in `Other changes`.

File: NEWS.md [hunk 3]
```diff
@@ -1,133 +1,203 @@
...
+     values, while the JSON array `names` would contain the associated names, in
+     the same order.
 
-## Highlights
+     `$(format-json values=list($(values .SDATA.*)) names=list($(names .SDATA.*)))`
+
+   The resulting name-value pairs are always sorted by their key, regardless of
+   the argument order.
+   ([#3911](https://github.com/syslog-ng/syslog-ng/pull/3911))
+ * `tls()` block: added option for restricting TLS 1.3 ciphers
```
Comment (MrAnno) [inline NEWS.md:27]: Optional/opinion: `cipher-suite(tls13())` is not really a big deal. I mean, we obviously need an option to restrict TLS 1.3 ciphers, but it was more of a bug that we forgot to add this in the first place, so I would not list it as a highlight.

The `rename()` rewrite rule or `symlink-as()` however seems to be a good highlight candidate in my opinion.

**Review by MrAnno — APPROVED** [review]

**Comment by MrAnno** [comment]
I think no news entry is needed for the JSON typing PR for v3.36 as it is a pre4.0 feature, which will be introduced as a set of blog posts by @bazsi.

**Comment by gaborznagy** [comment]
> I think no news entry is needed for the JSON typing PR as it is a pre4.0 feature, which will be introduced as a set of blog posts by @bazsi.

Not even a one line entry?

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: NEWS.md (hunk 2 above)
Comment (alltilla) [reply NEWS.md]: Fixed!

File: NEWS.md (hunk 3 above)
Comment (alltilla) [reply NEWS.md]: Thanks, fixed both.

**Review by MrAnno — APPROVED** [review]

**Comment by MrAnno** [comment]
> Not even a one-line entry?

Writing a one-line entry about just the JSON typing is not really useful in my opinion, because it is a `@version: 4.0` feature.
If we wanted to announce anything about the pre4.0 features in the news file, we would need to start with explaining the pre4.0 config version, how it can be enabled, etc. I'd absolutely prefer doing it, we could also reference Bazsi's blog posts about this topic, as it is a topic that requires a longer explanation. We can also do that in the next 3.37 news file.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**